src/consensus.rs

```diff
@@ -0,0 +1,125 @@
+use std::{
+    sync::mpsc::{self, Receiver, RecvTimeoutError, Sender},
+    time::{Duration, Instant},
+};
+
+use raft::{prelude::*, storage::MemStorage};
+
+pub struct Consensus {
+    node: RawNode<MemStorage>,
+    receiver: Receiver<Message>,
+}
+
+impl Consensus {
+    pub fn new(logger: &slog::Logger) -> (Self, Sender<Message>) {
+        let config = Config {
+            id: 1,
+            ..Default::default()
+        };
+        config.validate().unwrap();
+        let storage = MemStorage::new_with_conf_state((vec![1], vec![]));
+        let node = RawNode::new(&config, storage, logger).unwrap();
+        let (sender, receiver) = mpsc::channel();
+        (Self { node, receiver }, sender)
+    }
+
+    pub fn start(&mut self) {
+        let mut t = Instant::now();
+        let mut timeout = Duration::from_millis(100);
+
+        loop {
+            match self.receiver.recv_timeout(timeout) {
+                Ok(message) => self.node.step(message).unwrap(),
+                Err(RecvTimeoutError::Timeout) => (),
+                Err(RecvTimeoutError::Disconnected) => return,
```

Comment:
maybe we need to log something or return a proper error here?
I am worried about the lack of visibility during future debugging session.

Reply:
Here the timeout is used for periodic ticks - basically when it timeouts - we do process some data periodically. If it returns a message - we process a message and do our usual periodic update. When the sender is dropped - we receive `Disconnected`  err - but this will only happen on shutdown now. Do you think we should add a log there for `Disconnected` case? Like `Stopping Raft thread`?